src/impl_ops.rs

```diff
@@ -179,7 +179,13 @@ where
 {
     type Output = Array<A, <D as DimMax<E>>::Output>;
     fn $mth(self, rhs: &'a ArrayBase<S2, E>) -> Self::Output {
-        let (lhs, rhs) = self.broadcast_with(rhs).unwrap();
+        let (lhs, rhs) = if self.ndim() == rhs.ndim() && self.shape() == rhs.shape() {
+            let lhs = self.to_dimensionality::<<D as DimMax<E>>::Output>().unwrap();
```

Comment:
These to_dimensionality calls seem redundant with the ones already in `broadcast_with`. Is there a reason this PR shouldn't just do one or the other, not both?

If it was just the calls in this method, here it looks like `self.view().into_dimensionality::<...>()` is enough, i.e. using the existing method.

Reply:
That's right. We should use view().into_dimension() here. It is faster in the benchmark test (though I don’t know why)

Reply:
Adding the same shape detection in impl_ops can avoid executing co_broadcast. I think it has reserved value.